CMake: configuration fails for BUILD_SHARED_LIBS=OFF

When running:
```
git clone git@github.com:dealii/dealii.git
mkdir build
cd build
cmake -DBUILD_SHARED_LIBS=OFF ../dealii
```

We get the following error:

```
-- Configuring done
CMake Error: install(EXPORT "deal.IITargets" ...) includes target "dealii_debug" which requires target "bundled_kokkos_debug" that is not in any export set.
CMake Error: install(EXPORT "deal.IITargets" ...) includes target "dealii_debug" which requires target "bundled_tbb_debug" that is not in any export set.
CMake Error: install(EXPORT "deal.IITargets" ...) includes target "dealii_debug" which requires target "bundled_umfpack_L_UMF_debug" that is not in any export set.
CMake Error: install(EXPORT "deal.IITargets" ...) includes target "dealii_debug" which requires target "bundled_umfpack_Z_UMF_debug" that is not in any export set.
```

Full output: [configure.log](https://github.com/dealii/dealii/files/11030745/configure.log) ,[detailed.log](https://github.com/dealii/dealii/files/11030749/detailed.log)

This worked prior to #14491 

fyi @c-p-schmidt

--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -83,7 +83,7 @@ foreach(build ${DEAL_II_BUILD_TYPES})
   #
 
   set(_object_files "")
-  if(CMAKE_SYSTEM_NAME MATCHES "Windows")
+  if(NOT BUILD_SHARED_LIBS)
     get_property(_object_targets GLOBAL PROPERTY DEAL_II_OBJECT_TARGETS_${build})
     foreach(_target ${_object_targets})
       list(APPEND _object_files "$<TARGET_OBJECTS:${_target}>")
@@ -99,7 +99,7 @@ foreach(build ${DEAL_II_BUILD_TYPES})
   #
   populate_target_properties(${DEAL_II_NAMESPACE}_${build_lowercase} ${build})
 
-  if(NOT CMAKE_SYSTEM_NAME MATCHES "Windows")
+  if(BUILD_SHARED_LIBS)
     #
     # Add all object targets as private link targets
     #

